src/host/os_getnumcpus.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * \file   os_getcwd.c
+ * \brief  Retrieve the current working directory.
+ * \author Copyright (c) 2002-2008 Jason Perkins and the Premake project
+ */
+
+#include "premake.h"
+
+#if PLATFORM_LINUX
+#include <sched.h>
```

Comment:
Would it be simpler to `#define _GNU_SOURCE` here?
It even seems that `_POSIX_C_SOURCE=200809L` or `_XOPEN_SOURCE=700` are more appropriate.